server/src/test/java/org/elasticsearch/cluster/service/MasterServiceTests.java

```diff
@@ -2121,6 +2121,56 @@ public void onFailure(Exception e) {
         }
     }
 
+    public void testTimeoutRejectionBehaviourAtSubmission() {
+
+        final var actionCount = new AtomicInteger();
+
+        final var deterministicTaskQueue = new DeterministicTaskQueue();
+        final var threadPool = deterministicTaskQueue.getThreadPool(r -> {
+            if (r.toString().startsWith("master service timeout handler for [batched][test task]")) {
+                actionCount.incrementAndGet();
+                throw new EsRejectedExecutionException("simulated rejection", true);
+            } else {
+                return r;
+            }
+        });
+
+        try (var masterService = createMasterService(true, null, threadPool, new StoppableExecutorServiceWrapper(threadPool.generic()))) {
+
+            class TestTask implements ClusterStateTaskListener {
+                @Override
+                public void onFailure(Exception e) {
+                    if ((e instanceof FailedToCommitClusterStateException
```

Comment:
"Only count errors that exactly match the failure we inject above. Any others are unexpected."

Reply:
Is that a note-to-self or a suggested comment? :)

Reply:
Ha, sorry: suggested comment